cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -97,6 +112,55 @@ public static void validateConfig(SecureStore secureStore, SourceControlConfig s
     }
   }
 
+  // TODO CDAP-20343, Implement RepositoryManager#getFileHash()
+  public String getFileHash(Path filePath) throws IOException, NotFoundException {
+    return "place holder for fileHash";
+  }
+
+  /**
+   * Commits and pushes all changes under the repository base path
+   * @param commitMeta Details for the commit including author, committer and commit message
+   * @throws UnexpectedRepositoryChangesException when there are changes outside the repository base path
+   * @throws GitAPIException when the underlying git commands fail
+   * @throws AuthenticationConfigException when correct authentication credentials are not available.
+   */
+  public void commitAndPush(CommitMeta commitMeta) throws UnexpectedRepositoryChangesException,
+    GitAPIException, AuthenticationConfigException {
+    validateInitialized();
+
+    // if the status is clean skip
+    Status preStageStatus = git.status().call();
+    if (preStageStatus.isClean()) {
+      return;
+    }
+
+    // We should only add files inside the base path. There should not be any change outside the base path.
+    git.add().addFilepattern(getGitAddRelativePath()).call();
```

Comment:
Will this work well with symbolic links? Consider a case when the link is inside the pathPrefix, but the target is not. I guess the operationRunner will write to the target, so the changes inside the pathPrefix will be empty. We should have a test for this.

Reply:
Correct. I think its better to add individual files. Will make the necessary changes